runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -1172,6 +1182,39 @@ ClassFileOracle::walkMethodCodeAttributeAttributes(U_16 methodIndex)
 							_buildResult = GenericError;
 							break;
 						}
+
+						J9CfrLocalVariableTypeTableEntry lvttEntry = localVariableTypeTable->localVariableTypeTable[localVariableTypeTableIndex];
+						UDATA result = FALSE;
+						/* There may be more than one LocalVariableTable in the code attribute's attribute table. Check each attribute to find the match */
+						for (U_16 attributeIndex2 = 0; attributeIndex2 < attributesCount; ++attributeIndex2) {
+							J9CfrAttribute *codeAttrib = codeAttribute->attributes[attributeIndex2];
+							if(CFR_ATTRIBUTE_LocalVariableTable == codeAttrib->tag) {
+								for(U_16 lvtIndex = 0; lvtIndex < ((J9CfrAttributeLocalVariableTable*)codeAttrib)->localVariableTableLength; lvtIndex++) {
```

Comment:
Given that the index has been assigned at "U_16 index = localVariableTypeTable->localVariableTypeTable[localVariableTypeTableIndex].index;" on the top of your changes, I don't think a nested loop is necessary to check the pair. The idea is first check the localVariableTable based on the given index to see whether the entry exists. If yes, then check the pair.

Reply:
The spec implies that it is possible to have more than one LocalVariableTable attribute in a single Code attribute. Because of that it may not always be correct to assume that the position of the J9CfrLocalVariableTableEntry in the table matches the index value.

> If multiple LocalVariableTable attributes are present in the attributes table of
a Code attribute, then they may appear in any order.

Reply:
In such case, is it possible to take advantage of `_methodsInfo[methodIndex].localVariablesInfo[index].localVariableTable` (non-NULL) to deal with values in `J9CfrLocalVariableTableEntry`?   Normally LocalVariableTable should be addressed before LocalVariableTypeTable.

Reply:
The code looks like:
```
 if  (_methodsInfo[methodIndex].localVariablesInfo[index].localVariableTable != NULL ) {
    // it means the same index already exists in localVariableTableEntry
   //so compare the all values (including the index) in localVariableTableEntry in a single for loop
    // (the mismatch on index should be captured here in most cases)
    J9CfrLocalVariableTableEntry* LVTEntries = _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTable->localVariableTable;
   int LVTLength = _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTable->localVariableTableLength;
bool  entryFound = false;
    for (i = 0;  i < LVTLength; i++) {
            if ((LVTTentry.index == LVTEntries[i].index)
             && (LVTTentry.startPC== LVTEntries[i].startPC)
            && (LVTTentry.length == LVTEntries[i].length)
            && (LVTTentry.nameIndex== LVTEntries[i].nameIndex)) {
                      entryFound = true;
                      break;
           }
     }
   if (entryFound == false) { throw out ClassFormatError }
} else {
  // the _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTable might be not yet filled up.
  place your code here to do a full comparison of entries in LVT and LVTT tables.
}
```

Reply:
good idea.

Reply:
for the full comparison i can also start searching at the current point in the code attribute table instead of starting at 0.